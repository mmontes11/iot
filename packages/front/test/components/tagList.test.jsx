import React from "react";
import { shallow } from "enzyme";
import TagList from "components/tagList";

describe("components/tagList", () => {
  it("renders nothing when no tags are specified", () => {
    const wrapper = shallow(<TagList label="foo" tagStyle="is-warning" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders tags", () => {
    const wrapper = shallow(<TagList tags={["foo", "bar"]} label="foo" tagStyle="is-warning" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders tags without label", () => {
    const wrapper = shallow(<TagList tags={["foo", "bar"]} tagStyle="is-warning" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders tags without style", () => {
    const wrapper = shallow(<TagList tags={["foo", "bar"]} label="foo" />);
    expect(wrapper).toMatchSnapshot();
  });
});
