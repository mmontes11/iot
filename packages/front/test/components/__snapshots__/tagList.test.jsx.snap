// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/tagList renders nothing when no tags are specified 1`] = `""`;

exports[`components/tagList renders tags 1`] = `
<div
  className="tag-list"
>
  <p
    className="tag-list-label is-6"
  >
    <FormattedMessage
      id="foo"
      values={Object {}}
    >
      <Component />
    </FormattedMessage>
  </p>
  <div
    className="tags"
  >
    <FormattedMessage
      defaultMessage="foo"
      id="foo"
      key="foo"
      values={Object {}}
    >
      <Component />
    </FormattedMessage>
    <FormattedMessage
      defaultMessage="bar"
      id="bar"
      key="bar"
      values={Object {}}
    >
      <Component />
    </FormattedMessage>
  </div>
</div>
`;

exports[`components/tagList renders tags without label 1`] = `
<div
  className="tag-list"
>
  <div
    className="tags"
  >
    <FormattedMessage
      defaultMessage="foo"
      id="foo"
      key="foo"
      values={Object {}}
    >
      <Component />
    </FormattedMessage>
    <FormattedMessage
      defaultMessage="bar"
      id="bar"
      key="bar"
      values={Object {}}
    >
      <Component />
    </FormattedMessage>
  </div>
</div>
`;

exports[`components/tagList renders tags without style 1`] = `
<div
  className="tag-list"
>
  <p
    className="tag-list-label is-6"
  >
    <FormattedMessage
      id="foo"
      values={Object {}}
    >
      <Component />
    </FormattedMessage>
  </p>
  <div
    className="tags"
  >
    <FormattedMessage
      defaultMessage="foo"
      id="foo"
      key="foo"
      values={Object {}}
    >
      <Component />
    </FormattedMessage>
    <FormattedMessage
      defaultMessage="bar"
      id="bar"
      key="bar"
      values={Object {}}
    >
      <Component />
    </FormattedMessage>
  </div>
</div>
`;
